darabonba-openapi/client/client.go

```diff
@@ -0,0 +1,2146 @@
+			request_.Headers = dara.Merge(map[string]*string{
+				"host":          client.Endpoint,
+				"x-acs-version": version,
+				"x-acs-action":  action,
+				"user-agent":    openapiutil.GetUserAgent(client.UserAgent),
+			}, globalHeaders,
+				extendsHeaders,
+				request.Headers)
+		} else {
+			request_.Headers = dara.Merge(map[string]*string{
+				"host":          client.Endpoint,
+				"x-acs-version": version,
+				"x-acs-action":  action,
+				"user-agent":    openapiutil.GetUserAgent(client.UserAgent),
+			}, globalHeaders,
+				extendsHeaders,
+				request.Headers,
+				headers)
+		}
+
+		if !dara.IsNil(request.Body) {
+			m := dara.ToMap(request.Body)
+			tmp := dara.ToMap(openapiutil.Query(m))
+			request_.Body = dara.ToReader(dara.ToFormString(tmp))
+			request_.Headers["content-type"] = dara.String("application/x-www-form-urlencoded")
+		}
+
+		if dara.StringValue(authType) != "Anonymous" {
+			if dara.IsNil(client.Credential) {
+				_err = &ClientError{
+					Code:    dara.String("InvalidCredentials"),
+					Message: dara.String("Please set up the credentials correctly. If you are setting them through environment variables, please ensure that ALIBABA_CLOUD_ACCESS_KEY_ID and ALIBABA_CLOUD_ACCESS_KEY_SECRET are set correctly. See https://help.aliyun.com/zh/sdk/developer-reference/configure-the-alibaba-cloud-accesskey-environment-variable-on-linux-macos-and-windows-systems for more details."),
+				}
+				if dara.BoolValue(client.DisableSDKError) != true {
+					_err = dara.TeaSDKError(_err)
+				}
+				return _result, _err
+			}
+
+			credentialModel, _err := client.Credential.GetCredential()
+			if _err != nil {
+				retriesAttempted++
+				retryPolicyContext = &dara.RetryPolicyContext{
+					RetriesAttempted: retriesAttempted,
+					HttpRequest:      request_,
+					HttpResponse:     response_,
+					Exception:        _err,
+				}
+				_resultErr = _err
+				continue
+			}
+
+			if !dara.IsNil(credentialModel.ProviderName) {
+				request_.Headers["x-acs-credentials-provider"] = credentialModel.ProviderName
+			}
+
+			credentialType := dara.StringValue(credentialModel.Type)
+			if credentialType == "bearer" {
+				bearerToken := dara.StringValue(credentialModel.BearerToken)
+				request_.Query["BearerToken"] = dara.String(bearerToken)
+				request_.Query["SignatureType"] = dara.String("BEARERTOKEN")
+			} else if credentialType == "id_token" {
+				idToken := dara.StringValue(credentialModel.SecurityToken)
+				request_.Headers["x-acs-zero-trust-idtoken"] = dara.String(idToken)
+			} else {
+				accessKeyId := dara.StringValue(credentialModel.AccessKeyId)
+				accessKeySecret := dara.StringValue(credentialModel.AccessKeySecret)
+				securityToken := dara.StringValue(credentialModel.SecurityToken)
+				if !dara.IsNil(dara.String(securityToken)) && securityToken != "" {
+					request_.Query["SecurityToken"] = dara.String(securityToken)
+				}
+
+				request_.Query["SignatureMethod"] = dara.String("HMAC-SHA1")
```

Comment:
Rad flag for Security Team, it looks like weak algorithm

Reply:
Not sure if changing this one would actually break how the module connects to the remote API.